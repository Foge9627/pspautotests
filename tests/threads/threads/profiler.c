#include "common.h"
#include "shared.h"
#include "string.h"

int main(int argc, char *argv[]) {
	CHECKPOINT_ENABLE_TIME = 1;
	int i;

	checkpointNext("sceKernelReferThreadProfiler");
	for (i = 0; i < 8; i++) {
		PspDebugProfilerRegs *regs = sceKernelReferThreadProfiler();
		checkpoint("regs: %08x", regs);
	}

	checkpointNext("sceKernelReferGlobalProfiler");
	for (i = 0; i < 8; i++) {
		PspDebugProfilerRegs *regs = sceKernelReferGlobalProfiler();
		checkpoint("regs: %08x", regs);
	}
	return 0;
}
