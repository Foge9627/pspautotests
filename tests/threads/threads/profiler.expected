[x/183566489] sceKernelReferThreadProfiler
[x/92] regs: 00000000
[x/65] regs: 00000000
[x/57] regs: 00000000
[x/57] regs: 00000000
[x/56] regs: 00000000
[x/58] regs: 00000000
[x/58] regs: 00000000
[x/57] regs: 00000000

[x/4608] sceKernelReferGlobalProfiler
[x/70] regs: 00000000
[x/64] regs: 00000000
[x/57] regs: 00000000
[x/57] regs: 00000000
[x/57] regs: 00000000
[x/56] regs: 00000000
[x/57] regs: 00000000
[x/55] regs: 00000000
